lib/spack/spack/operating_systems/cle.py

```diff
@@ -80,3 +84,54 @@ def find_compiler(self, cmp_cls, *paths):
                 compilers.append(comp)
 
         return compilers
+
+
+class CrayFrontend(LinuxDistro):
+    """Represents OS that runs on login and service nodes of the Cray
+    platform. It acts as a regular Linux without Cray-specific modules and
+    compiler wrappers."""
+
+    def find_compilers(self, *paths):
+        """Calls the overridden method but prevents it from detecting Cray
+        compiler wrappers to avoid possible false detections. The detected
+        compilers come into play only if a user decides to work with the
+        Cray's frontend OS as if it was a regular Linux environment."""
+
+        environ_bu = None
+
+        # We rely on the fact that the PrgEnv-* modules set the PE_ENV
+        # environment variable.
+        if 'PE_ENV' in os.environ:
+
+            # Copy environment variables to restore them after the compiler
+            # detection. We expect that the only thing PrgEnv-* modules do is
+            # the environment variables modifications.
+            environ_bu = os.environ.copy()
+
+            # Restore the name of the module from the environment variable.
+            prg_env = 'PrgEnv-' + os.environ['PE_ENV'].lower()
+
+            modulecmd = which('modulecmd', required=True)
```

Comment:
Sure, a lot has changed since I submitted this PR. I will try to come with an updated version tomorrow.